ORG-187 Add documentation for Organization Domains and <OrganizationList /> (#271)
clerkinc/clerk-docs: docs/components/organization/organization-list.mdx

        {...}
    />
  );
};
```

## Props

All props below are optional.

| Name | Type | Description |
| --- | --- | --- |
| `hidePersonal` | `boolean` | By default, users can switch between organization and their personal account. This option controls whether `OrganizationList` will include the user's personal account in the organization list. Setting this to `false` will hide the personal account entry, and users will only be able to switch between organizations.<br />Defaults to `false`.
| `skipInvitationScreen` | `boolean` \| `undefined` | Hides the screen for sending invitations after an organization is created. When left undefined Clerk will automatically hide the screen if the number of max allowed members is equal to 1.<br />Defaults to `false`.
| `apperance` | `object` | Controls the overall look and feel.
| `afterCreateOrganizationUrl` | `((org: Organization) => string)` \| `string` | Full URL or path to navigate after creating a new organization.<br />Defaults to `undefined`.
| `afterSelectOrganizationUrl` | `((org: Organization) => string)` \| `string` | Full URL or path to navigate after selecting an organization.<br />Defaults to `undefined`.

> | `afterSelectPersonalUrl` | `((org: Organization) => string)` \| `string` | Full URL or path to navigate after selecting the personal account.<br />Defaults to `undefined`.

Reviewer: the above did not work, i went with 
```suggestion
| `afterSelectPersonalUrl` | <code>((org: [Organization][org-ref]) => string)</code> \| `string` | Full URL or path to navigate after selecting the personal account.<br />Defaults to `undefined`.
```

---
title: <OrganizationList />

> description: The `<OrganizationList />` component is used to display organization related memberships, invitations, and suggestions for the user.

Reviewer 2: We can remove MDX from the SEO description.

```suggestion
description: The <OrganizationList /> component is used to display organization related memberships, invitations, and suggestions for the user.
```

---

# `<OrganizationList />`

<Images
  alt="Organization List Example"
  width={992}
  height={568}
  src="/docs/images/ui-components/component-org_list.svg"

> />

Reviewer: @mzhong9723  Does this image exist ? I couldn't get it to work locally

Author: Added image as a PNG, not sure if we should be using a SVG to be consistent - if so, do we usually export those from Figma?

Reviewer 2: Left a review comment, but these should be optimized, preferably as WebP.
I use Squoosh, and the default WebP settings.

The `<OrganizationList />` component is used to display organization related memberships, invitations, and suggestions for the user.

## Usage

<Tabs items={['Next.js', 'React', 'Remix']}>
  <Tab>

>     <CodeBlockTabs options={["App Router", "Pages Router"]}>

Reviewer 2: Let's use CodeBlockTabs instead of Tabs, and we can have the options be {["Next.js App Router", "Next.js Pages Router", "React", "Remix"]}

Tabs should be more for complex content.

                afterSelectOrganizationUrl={org => `/organization/${org.slug}`}
            />
        );
      }
      ```
    </CodeBlockTabs>
  </Tab>

  <Tab>
    ```jsx copy filename="discover.tsx"
    import { OrganizationList } from "@clerk/clerk-react";

    export default function OrganizationListPage() {
      return (
          <OrganizationList
              afterCreateOrganizationUrl={org => `/organization/${org.slug}`}
              afterSelectPersonalUrl={user => `/user/${user.id}`}
              afterSelectOrganizationUrl={org => `/organization/${org.slug}`}
          />
      );
    }
    ```
  </Tab>

  <Tab>
    ```jsx copy filename="/route/discover.tsx"
    import { OrganizationList } from "@clerk/remix";

    export default function OrganizationListPage() {
      return (
          <OrganizationList
              afterCreateOrganizationUrl={org => `/organization/${org.slug}`}
              afterSelectPersonalUrl={user => `/user/${user.id}`}
              afterSelectOrganizationUrl={org => `/organization/${org.slug}`}
          />
      );
    }
    ```
  </Tab>
</Tabs>

### Force Organizations

> For specific applications, we want to prohibit people from using their personal accounts as they need to be part of an organization.
> Use <code>{"<OrganizationList/>"}</code> with <code>{"hidePeronsal"}</code> in order to guide your user to join or create an organization in order to continue.

Reviewer 2: ```suggestion
If you would like to prohibit people from using their personal accounts and force them to be part of an organization, the `hidePersonal` property forces your user to join or create an organization in order to continue.
```

```tsx copy filename="organization-portal.tsx"
export default function OrganizationListPage() {
  return (
    <OrganizationList
        hidePersonal={true}
        {...}
    />
  );
};
```

## Props

All props below are optional.

> <Tables

Reviewer 2: This table does not require complex use of JSX components, so let's use a markdown table.

| Name | Type | Description |
| --- | --- | --- |
| `hidePersonal` | `boolean` | By default, users can switch between organization and their personal account. This option controls whether `OrganizationList` will include the user's personal account in the organization list. Setting this to `false` will hide the personal account entry, and users will only be able to switch between organizations.<br />Defaults to `false`.

etc etc etc
